git-extra/git-credential-chooser.c

```diff
@@ -10,20 +11,99 @@ static size_t helper_nr, selected_helper;
 
 static int discover_helpers(void)
 {
-	/* TODO: walk `PATH` to populate these */
-	helper_nr = 2;
-	helper_name = malloc(helper_nr * sizeof(*helper_name));
-	helper_path = malloc(helper_nr * sizeof(*helper_path));
+	WCHAR *pattern_suffix = L"\\git-credential-*";
+	size_t alloc = 16, env_size, pattern_len, pattern_suffix_len = wcslen(pattern_suffix);
+	WCHAR env[32768], pattern[32768 + pattern_suffix_len], *p;
+	HANDLE h;
+	WIN32_FIND_DATAW find_data;
+
+	env_size = GetEnvironmentVariableW(L"PATH", env, 32768);
```

Comment:
DRY: use `sizeof env` instead of the number.

Reply:
`sizeof(env)` is sadly double the number of characters (`env` is a Unicode string). Besides, in this case I really wanted to use the documented maximal length of an environment variable. Do you know any constant for that?

Reply:
Aha! It's `_MAX_ENV`: https://docs.microsoft.com/en-us/cpp/c-runtime-library/max-env?view=vs-2019